test/unit_test/workflows/test_run_config.py

```diff
@@ -54,26 +54,53 @@ def test_config_without_azureml_config(self):
             RunConfig.parse_obj(user_script_config)
             assert str(e.value) == "AzureML client config is required for AzureML system"
 
+    @patch("azure.identity._credentials.default.InteractiveBrowserCredential")
+    @patch("azure.identity._credentials.default.AzurePowerShellCredential")
+    @patch("azure.identity._credentials.default.AzureCliCredential")
```

Comment:
We need to mock all the credentials because the default credential will get tokens from all of them. Remove some duplicated assertions.  Keep several to make sure olive azureml client config can append appropriated credentials.